metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/service/dao/WikidataAccessDao.java

```diff
@@ -138,47 +166,84 @@ public WikidataOrganization parse(String xml) throws JAXBException {
    * @param uri The Wikidata URI in string format
    * @return RDF model
    */
-  private Model getModelFromSPARQL(String uri) {
+  private Resource getModelFromSPARQL(String uri) {
+    Resource resource = fetchFromSPARQL(uri);
+    if (!isDuplicate(resource)) {
```

Comment:
I improved exception handling, fetchFromSparql is not returning null anymore, but it throws exception. isDupplicate(resource) was updated to avoid NPE